src/asm3/stock.py

```diff
@@ -122,6 +324,186 @@ def update_stocklevel_from_form(dbo: Database, post: PostedData, username: str)
     if diff != 0: 
         insert_stockusage(dbo, username, slid, diff, post.date("usagedate"), post.integer("usagetype"), post["comments"])
 
+def insert_product_from_form(dbo: Database, post: PostedData, username: str) -> int:
+    """
+    Inserts a product item from a dialog.
+    """
+    l = dbo.locale
+    if post["productname"] == "":
+        raise asm3.utils.ASMValidationError(_("Product must have a name", l))
+    
+    retired = 0
+    if post.integer("active") == 0:
+        retired = 1
+   
+    pid = dbo.insert("product", {
+        "ProductName":          post["productname"],
+        "Description":          post["productdescription"],
+        "ProductTypeID":        post.integer("producttypeid"),
+        "SupplierID":           post.integer("supplierid"),
+        "UnitTypeID":           post.integer("unittypeid"),
+        "CustomUnit":           post["customunit"],
+        "PurchaseUnitTypeID":   post.integer("purchaseunittypeid"),
+        "CustomPurchaseUnit":   post["custompurchaseunit"],
+        "CostPrice":            post.integer("costprice"),
+        "RetailPrice":          post.integer("retailprice"),
+        "UnitRatio":            post.integer("unitratio"),
+        "TaxRateID":            post.integer("taxrateid"),
+        "IsRetired":            retired,
+        "Barcode":              post["barcode"],
+        "PLU":                  post["plu"],
+        "GlobalMinimum":        post["globalminimum"],
+        "RecentBatchNo":        "",
+        "RecentExpiry":         ""
+    }, username)
+
+    return pid
+
+def insert_productmovement_from_form(dbo: Database, post: PostedData, username: str) -> int:
+    """
+    Inserts a product movement from a dialog.
+    """
+    movementusagetypeid = asm3.configuration.product_movement_usage_type(dbo)
+    l = dbo.locale
+    if post["movementdate"] == "":
+        raise asm3.utils.ASMValidationError(_("Movement must have a date", l))
+    
+    locations = []
+
+    if post.integer("movementfromtype") == 0 and post.integer("movementtotype") == 1:# Stock to usage
+        locations = [post.integer("movementfrom"), False]
+        usagetypeid = post.integer("movementto")
+    elif post.integer("movementfromtype") == 0 and post.integer("movementtotype") == 0:# Stock to stock
+        locations = [post.integer("movementfrom"), post.integer("movementto")]
+        usagetypeid = movementusagetypeid
+    elif post.integer("movementfromtype") == 1 and post.integer("movementtotype") == 0:# Usage to stock
+        locations = [False, post.integer("movementto"),]
```

Comment:
I'm not finding using a 2 element list of locations very readable? Wouldn't this be a lot simpler written as two variables, fromlocation and tolocation, with 0 indicating no location/usage type?